hippylib/modeling/timeDependentVector.py

```diff
@@ -131,14 +138,66 @@ def norm(self, time_norm, space_norm):
             tmp = self.data[i].norm(space_norm)
             if tmp > s_norm:
                 s_norm = tmp
-        
+
         return s_norm
 
     def inner(self, other):
         """
         Compute the inner products: :math:`a+= (\\mbox{self[i]},\\mbox{other[i]})` for each snapshot.
         """
-        a = 0.
+        assert (
+            other.nsteps == self.nsteps
+        ), "vectors do not have the same number of snapshots"
+        a = 0.0
         for i in range(self.nsteps):
             a += self.data[i].inner(other.data[i])
-        return a
\ No newline at end of file
+        return a
+
+    def element_wise_inner(self, other):
```

Comment:
The name of this function is odd. If I understand correctly here you compute

out(t) = int_omega a(x,t) b(x,t) dx.

But in a discrete settings?

Reply:
Yes... this was supposed to be a helper function to compute L^2(0,T;\Omega) norm when you don't have a uniform time-discretization. The idea being that you'd multiply a copy of the TD vector with the mass matrix `Mz`, and then call `z.element_wise_inner(Mz)`. For fixed time-stepping, `inner` is fine and you can pre-multiply by `1/time_step`.

This is a pretty specific use-case, so I'd be fine dropping support if you prefer.